src/service_daemon.rs

```diff
@@ -1671,14 +1702,15 @@ fn call_listener(
 }
 
 /// Returns valid IPv4 interfaces in the host system.
-fn my_ipv4_interfaces() -> Vec<Ifv4Addr> {
+fn my_ipv4_interfaces(interface: Option<&str>) -> Vec<Ifv4Addr> {
```

Comment:
With this change, I realized that the name `my_ipv4_interfaces` seems not very accurate because it returns only the address info.  How about renaming it to `my_ipv4_addrs` ?